### ISSUE TITLE
feat(rules,html-spec): enforce three contextual ARIA constraints (popovertarget+aria-expanded, input[type=hidden]+aria-hidden, summary[role] in details)

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3735]
## Summary

Enforce three conditional ARIA constraints not currently detected by markuplint:

1. `<input type="hidden">` must not have `aria-hidden` attribute
2. `<button popovertarget>` must not have `aria-expanded` attribute (the popover API manages state)
3. `<summary role>` inside `<details>` is prohibited

## Background

Split from #3633 (closed). The original issue listed 9 patterns; verification showed 6 were already detected via `wai-aria-disallowed-props` + `properties.without`. Only these 3 patterns remain truly missed.

### Pattern 1: input[type=hidden] aria-hidden

```html
<!-- Invalid -->
<input type="hidden" aria-hidden="true">
```

### Pattern 2: button[popovertarget] aria-expanded

```html
<!-- Invalid: popover API manages expanded state automatically -->
<button popovertarget="p" aria-expanded="false">Toggle</button>
```

### Pattern 3: summary[role] inside details

```html
<!-- Invalid: summary inside details has fixed semantics -->
<details>
  <summary role="button">Toggle</summary>
  <p>Content</p>
</details>
```

## Spec reference

- https://w3c.github.io/html-aria/

## Verification

nu-html-checker confirmed output (via Docker):
- All 3 patterns report errors in nu
- markuplint currently reports 0 errors on all 3

## Implementation approach

Consider adding `properties.without` entries or `conditions` to relevant element spec data in `@markuplint/html-spec`:
- `input[type=hidden]` — add must-not for aria-hidden
- `button[popovertarget]` — add must-not for aria-expanded
- `summary` — add condition "inside details" that sets permittedRoles: false

## Affected packages

- `@markuplint/html-spec` — spec data
- Tests in `@markuplint/rules`

## Benchmark cross-reference

_Sync'd from `tests/external/snapshots/` (submodule=1429313, nu-validator@sha256:59b0e97e2664, markuplint=5.0.0-rc.4, generated=2026-04-27 06:10:57 UTC)._

Filter: `aria-hidden-on-input-hidden|popovertarget-with-aria-expanded|aria-expanded-with-popovertarget` → **3** fixtures. Verdict tally: match-error=0, match-clean=0, ml-only=0, nu-only=3, nu-over=0.

Two of three patterns (`input[type=hidden] aria-hidden`, `button[popovertarget] aria-expanded`) have `nu-only` fixtures and confirm the gap. `summary[role]` inside `<details>` has no fixture in the suite; keep the upstream manual check as the sole evidence for that third pattern.

| fixture | verdict | nu id / ml rule |
| --- | --- | --- |
| `html-aria/misc/aria-expanded-with-popovertarget-novalid.html` | nu-only | `nv-d020707a3552` |
| `html-aria/misc/aria-hidden-on-input-hidden-novalid.html` | nu-only | `nv-ea4600e92672` |
| `html/attributes/popovertarget-with-aria-expanded-novalid.html` | nu-only | `nv-7987209ad11d` |

Reproduce: see [`tests/external/CLAUDE.md`](https://github.com/markuplint/markuplint/blob/dev/tests/external/CLAUDE.md) "Auditing a claim against the benchmark".

**Comment by YusukeHirao** [comment]
## Docker 実測結果（2026-04-15）

### 全 3 ケース対比

| # | HTML | nu-html-checker | markuplint | Gap |
|---|------|----------------|-----------|-----|
| 1 | `<button popovertarget="p" aria-expanded="false">` | `The "aria-expanded" attribute must not be used on any element which has a "popovertarget" attribute.` | （検出なし） | **YES** |
| 2 | `<input type="hidden" aria-hidden="true">` | `The "aria-hidden" attribute must not be specified on an "input" element whose "type" attribute has the value "hidden".` | （検出なし） | **YES** |
| 3 | `<details><summary role="button">...</summary>...</details>` | warning: `The "button" role is unnecessary for element "summary".` + error: `The "role" attribute must not be used on any "summary" element that is a summary for its parent "details" element.` | （検出なし） | **YES** |

## 真のスコープ

3 件すべて完全漏れ。それぞれ実装場所が異なる:

### #1 popovertarget + aria-expanded
- 同一要素の cross-attribute 制約
- 修正箇所: `@markuplint/html-spec` の `<button>`（および popovertarget を受ける要素）の条件データで `popovertarget` 存在時の `properties.without` に `aria-expanded` を追加

### #2 input[type=hidden] + aria-hidden
- `type=hidden` 条件下での属性禁止
- 修正箇所: `@markuplint/html-spec` の `input[type=hidden]` condition に `properties.without: [{ name: "aria-hidden", type: "must-not" }]` 追加

### #3 summary[role] inside details
- 親要素文脈での role 禁止（`<details>` の直下 summary のみ）
- 修正箇所: `wai-aria-permitted-roles` のチェック関数で親要素判定、または `<summary>` の condition データで親 `<details>` 時に permittedRoles: false

## メッセージ形式（nu 準拠）

上記表の nu 出力をそのまま使用可能。

**Comment by YusukeHirao** [comment]
## 訂正: 各パターンの実装箇所判断が誤り

前コメントで各パターンの実装方針を「`properties.without` 追加」等と書いたが、コード確認で一部誤りが判明。以下が正しい現状と修正方針。

### パターン 1: `button[popovertarget] + aria-expanded`

- **現状**: `spec.button.jsonc` に `popovertarget` 属性定義はあるが ARIA condition は未設定
- **実測**: markuplint passed, nu: error
- **修正箇所**: `spec.button.jsonc` の aria section に `[popovertarget]` 条件を追加し、`properties.without` で `aria-expanded` を must-not に

### パターン 2: `input[type=hidden] + aria-hidden`

**前コメントの分析は誤り**: 「`properties: false` で block 済み」は実現していない。

- **spec データ**: `spec.input.jsonc` の `[type='hidden' i]` 条件で `properties: false` `implicitRole: false` `permittedRoles: false`
- **実装の欠陥**: `disallowed-prop.ts:81` が `if (!role) return;` で早期 return するため、`implicitRole: false` の要素に対して ARIA 属性チェックが走らない
- **これは #3630 naming prohibition と同じ構造**（解決済み Issue と共通の根本原因）
- **実測**: markuplint passed, nu: error
- **修正箇所**: `disallowed-prop.ts` で `elAriaSpec?.properties === false` AND `!role` かつ aria-* 属性が付与されている場合に違反として報告

### パターン 3: `<details><summary role="..."></details>`

**前コメントの分析は部分的に誤り**:

- **spec データ**: `spec.summary.jsonc` で `implicitRole: "button"`, `permittedRoles: false`
- **実測**:
  - `<summary role="button">` → markuplint **passed**（role が implicit と一致するため redundant 扱いで通る）
  - `<summary role="generic">` → markuplint detects（implicit と違うので検出）
- **nu の動作**: role が implicit と一致しても detail の要約としての summary には role attribute 自体を禁止
- **修正箇所**: 新しい parent-context check を `wai-aria-permitted-roles` または別サブルールに追加。条件: `<details>` の直下 `<summary>` のとき role attribute を禁止（redundant 警告ではなく明示的な禁止として）

### まとめ

3 パターンとも未実装で修正必要だが、実装箇所が異なる:

| # | 修正箇所 | 種類 |
|---|---------|------|
| 1 | `spec.button.jsonc` に条件付き properties.without | spec データ |
| 2 | `disallowed-prop.ts` で `properties === false` + `!role` を処理 | 共通チェック関数 |
| 3 | 新 check または既存 permitted-roles に parent context logic | ルール |
